Source/Core/Common/Config/Layer.cpp

```diff
@@ -38,56 +75,44 @@ Layer::~Layer()
   Save();
 }
 
-bool Layer::Exists(System system, const std::string& section_name, const std::string& key)
-{
-  Section* section = GetSection(system, section_name);
-  if (!section)
-    return false;
-  return section->Exists(key);
-}
-
-bool Layer::DeleteKey(System system, const std::string& section_name, const std::string& key)
+bool Layer::Exists(const ConfigLocation& location) const
 {
-  Section* section = GetSection(system, section_name);
-  if (!section)
-    return false;
-  return section->Delete(key);
+  const auto iter = m_map.find(location);
+  return iter != m_map.end() || iter->second.has_value();
```

Comment:
Shouldn't this be `iter != m_map.end() && iter->second.has_value()` instead of `||`?